backends/cuda-reg/ceed-cuda-reg-restriction.c

```diff
@@ -275,21 +204,23 @@ static int CeedElemRestrictionOffset_Cuda_reg(const CeedElemRestriction r,
   ierr = CeedElemRestrictionGetCeed(r, &ceed); CeedChk(ierr);
   CeedElemRestriction_Cuda_reg *impl;
   ierr = CeedElemRestrictionGetData(r, (void *)&impl); CeedChk(ierr);
-  CeedInt nelem, elemsize, nnodes;
+  CeedInt nelem, elemsize, lsize, ncomp;
```

Comment:
@YohannDudouit, @jedbrown I think I've found an issue with the `cuda/reg` and `cuda/shared` backends for mixed finite elements support. `cuda/ref` and `cuda/gen` are fine with the changes in this PR. 

It looks like knowledge of `nnodes` is being used heavily in the `cuda/reg` restrictions. This will break with mixed finite elements because `nnodes` does not make sense with a vector that holds a elements of different geometries, sizes, number of components, etc.

I can bully these backends into working against the current test suite, but I don't think we should merge something that will break.

So my questions are
1) Is it easy to remove this dependency on `nnodes`? @YohannDudouit, you'd know best how easy this is.
2) If it isn't easy, do we want to retire these backends (or at least temporarily) since they are not our high performance CUDA backend or our reference CUDA backend?

For context, I'm trying to get mixed finite element support added for someone who needs this code within a couple of weeks.

Reply:
At a glance, it looks like you already removed most/all of those references. Is that not working?

A stop-gap if you need it would be for the backend to convert offsets back into node indices (and fail with a helpful error message if that isn't possible).

Reply:
I did remove those references by assuming that `nnodes` is given by `lsize / ncomp`, but I am going to violate that assumption next week. I'm actually pretty surprised that this works for our composite operator tests.

Reply:
My suggestion (in lieu of input from others) would be to have the backend check that its assumptions are satisfied and error if not. That backend might not be usable for the mixed FEM spaces, but it can still pass the existing tests and serve as a platform if someone wants to revisit/optimize a backend with its strategy. I don't think we need to rush in deleting it.

Reply:
I've been thinking on this. I don't think that my conversion is very robust and it can't really be used as a check. But, I think that this use of `nnodes` is possible equivalent to `(max(offsets) - min(offsets)) / a_factor_that_depends_upon_ncomp_and_compstride`. I could try a one time computation of `max(offsets)` and `min(offsets)` to fix this in a more robust way.

Reply:
I fixed it, and hopefully its adequately robust for mixed finite elements.